### PR TITLE
TINKERPOP-1317: use graph class as root class to find temporary directory

### PR DESCRIPTION
**Pull request by pluradj**
https://issues.apache.org/jira/browse/TINKERPOP-1317

### CODE REVIEW

**Comment by dkuppitz** [comment]
VOTE: +1

**Comment by analytically** [comment]
Genius PR.

File: docs/src/dev/provider/index.asciidoc [hunk 1]
```diff
@@ -505,6 +505,11 @@ off for the test suite (the Maven SureFire Plugin is configured this way by defa
 include this setting, `<reuseForks>false</reuseForks>`, in the SureFire configuration if tests are failing in an
 unexplainable way.
 
+TIP: When running the `gremlin-test` suite against your implementation, you may need to set `build.dir` as an
+envrionment variable, depending on your project layout. Some tests require this to find a writable directory for
```
Comment (twilmes) [inline docs/src/dev/provider/index.asciidoc]: Tiny little typo here: "envrionment" -> "environment"

**Comment by spmallette** [comment]
@pluradj assuming @twilmes is voting +1 here pending the spelling fix, you will have your positive VOTE if you include a +1 yourself.

**Comment by twilmes** [comment]
Yes, looks good to me VOTE: +1

**Comment by pluradj** [comment]
I always forget to vote for my own stuff. Thanks all.

VOTE: +1

**Pull request merged**